### PR TITLE
iam auth plugin

### PR DESCRIPTION
**Pull request by bma13**

### CODE REVIEW

File: plugins/auth_nebius/auth_nebius.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+func processAuthenticateResponse(resp *pb.AuthenticateResponse) (string, auth.AuthCode, error) {
+	switch resp.ResultCode {
+	case pb.AuthenticateResponse_OK:
+		return accountToString(resp.Account), auth.AuthCodeSuccess, nil
+	case pb.AuthenticateResponse_UNKNOWN_SUBJECT:
+		return "", auth.AuthCodeUnknownSubject, nil
+	case pb.AuthenticateResponse_INVALID_TOKEN:
+		return "", auth.AuthCodeInvalidToken, nil
+	}
+	return "", auth.AuthCodeError, errors.New("unknown authaticate response result code")
```
Comment (qrort) [inline plugins/auth_nebius/auth_nebius.go:109]: typo

File: plugins/auth_nebius/auth_nebius.go [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+}
+
+func (p *authProviderNebius) Authenticate(ctx context.Context, token string) (string, auth.AuthCode, error) {
```
Comment (qrort) [inline plugins/auth_nebius/auth_nebius.go:112]: does this one have any usages?

File: cmd/ydbcp/main.go [hunk 3]
```diff
@@ -43,13 +48,34 @@ type server struct {
...
+	token = token[7:]
+	return token, nil
 }
 
 func (s *server) GetBackup(ctx context.Context, request *pb.GetBackupRequest) (*pb.Backup, error) {
 	xlog.Debug(ctx, "GetBackup", zap.String("request", request.String()))
 	requestId, err := types.ParseObjectId(request.GetId())
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse uuid %s: %w", request.GetId(), err)
+		return nil, fmt.Errorf("failed to parse uuid %s: %w", request.GetId(), err) // TODO: Permission denied?
```
[the remark this answers is not in the document]
Comment (bma13) [reply cmd/ydbcp/main.go]: you are right

File: cmd/ydbcp/main.go [hunk 4]
```diff
@@ -175,8 +210,44 @@ func (s *server) MakeBackup(ctx context.Context, req *pb.MakeBackupRequest) (*pb
...
+	if len(resourceID) > 0 {
+		checks[0].ResourceID = []string{resourceID}
+	}
+
+	resp, subject, err := s.auth.Authorize(ctx, token, checks)
+	if err != nil {
+		xlog.Error(ctx, "auth plugin authorize error", zap.Error(err))
+		return "", errPermissionDenied
+	}
+	if len(resp) != 1 {
```
[the remark this answers is not in the document]
Comment (bma13) [reply cmd/ydbcp/main.go:231]: there is checking plugin response, we can't rely on the plugin to return correct number of results

File: cmd/ydbcp/main.go [hunk 5]
```diff
@@ -64,18 +90,27 @@ func (s *server) GetBackup(ctx context.Context, request *pb.GetBackupRequest) (*
 		),
 	)
 	if err != nil {
-		xlog.Error(ctx, "can't select backups", zap.Error(err))
+		xlog.Error(ctx, "can't select backups", zap.Error(err)) // TODO: Permission denied?
```
[the remark this answers is not in the document]
Comment (bma13) [reply cmd/ydbcp/main.go]: no

**Review by qrort — APPROVED** [review]

**Pull request merged**